src/main/java/InfrastructureManager/ScenarioRunner.java

```diff
@@ -0,0 +1,78 @@
+package InfrastructureManager;
+
+/**
+ * Subclass of Runner specific to run scenarios
+ */
+public class ScenarioRunner extends Runner {
+
+    private Scenario scenario;
+    private final String scenarioName;
+    private int currentEvent;
+
+    /**
+     * Constructor of the class, has no input parameter because input will be the events in the
+     * scenario
+     * @param name Name of the runner
+     * @param scenarioName Name of the scenario that the runner is supposed to run
+     * @param outputs 1 or more MasterOutput objects to be defined as outputs of the runner
+     */
+    public ScenarioRunner(String name,String scenarioName, MasterOutput...outputs) {
+        super(name,null,outputs);
+        this.scenarioName = scenarioName;
+        this.scenario = null;
+        this.currentEvent = 0;
+    }
+
+    /**
+     * Set the scenario of the runner
+     * @param scenario Scenario to be run.
+     */
+    public void setScenario(Scenario scenario, long startTime) throws IllegalArgumentException{
+        this.scenario = scenario;
+        this.scenario.setStartTime(startTime);
+    }
+
+    /**
+     * Get the name of the scenario, this runner is to run according to configuration
+     * @return Name of the assigned scenario
+     */
+    public String getScenarioName() {
+        return scenarioName;
+    }
+
+
+    /**
+     * Overridden method in that goes in the run() method. Here, the execution of the master is
+     * specific to run events as inputs. A different event in the list is run, each time this
+     * method is called
+     */
+    @Override
+    public void runOperation() {
+        Event current = this.scenario.getEventList().get(currentEvent);
+        waitForEvent(current);
+        super.runOperation();
+        currentEvent++;
+        if (currentEvent == this.scenario.getEventList().size()) {
+            exit();
+        }
+    }
+
+    @Override
+    public void exit() {
+        currentEvent = 0;
+        super.exit();
+    }
+
+    /**
+     * Wait for executing the states according to their relative execution times
+     * @param e Event to be waited for
+     */
+    private void waitForEvent (Event e) {
+        while (true) {
```

Comment:
That is a busy waiting loop, the upside is that it gives highest precision possible, the downside is that it uses the CPU all of the time, which is not a good design choice when we expect a lot of these running at the same time.
You have the following solutions ordered by simplicity:
1. Add a little bit of delay each iteration (1 ms will do).
1. Calculate remaining time and sleep for exactly that time.
1. Same as above but sleep only half the period, repeating until the remaining time is short enough, then continue busy waiting (still with a short delay).

I prefer the second one, but you may do whichever you see fit.